include/nbl/builtin/hlsl/emulated_float64_t.hlsl

```diff
@@ -25,298 +29,312 @@
 #define FLOAT_ROUND_DOWN            2
 #define FLOAT_ROUND_UP              3
 #define FLOAT_ROUNDING_MODE         FLOAT_ROUND_NEAREST_EVEN
-
-namespace emulated
+namespace nbl
+{
+namespace hlsl
 {
-    namespace impl
+namespace impl
+{
+    template <typename T>
+    uint64_t promoteToUint64(T val)
     {
-        nbl::hlsl::uint32_t2 umulExtended(uint32_t lhs, uint32_t rhs)
-        {
-            uint64_t product = uint64_t(lhs) * uint64_t(rhs);
-            nbl::hlsl::uint32_t2 output;
-            output.x = uint32_t((product & 0xFFFFFFFF00000000) >> 32);
-            output.y = uint32_t(product & 0x00000000FFFFFFFFull);
-            return output;
-        }
-        
-        bool isNaN64(uint64_t val)
-        {
-            return bool((0x7FF0000000000000ull & val) && (0x000FFFFFFFFFFFFFull & val));
-        }
+        using AsFloat = unsigned_integer_of_size<sizeof(T)>;
+        uint64_t asUint = ieee754::impl::castToUintType(val);

```

Comment:
I think you might have a bug or the code is not readable at all

when you call this with `T=integer_type` you're probably reinterpreting it as a float or something

Reply:
irrevelant